Library/Homebrew/test/search_spec.rb

```diff
@@ -57,4 +58,33 @@
       end
     end
   end
+
+  describe "#search_descriptions" do
+    let(:args) { instance_double("args") }
```

Comment:
This line is failing `brew style`. `--fix` just removes the quotes which completely breaks the code. There should be an additional argument here of what class `args` is supposed to be, but that just gives a different error:
```
the Homebrew::CLI::Args class does not implement the instance method: formula?
the Homebrew::Cmd::Desc::Args class does not implement the instance method: formula?
```

Reply:
I was going to suggest using `OpenStruct` but it looks like we have a linter rule against that too. This works but is kind of hacky.

```diff
diff --git a/Library/Homebrew/test/search_spec.rb b/Library/Homebrew/test/search_spec.rb
index 11f5d2c9dc..afb409f2a9 100644
--- a/Library/Homebrew/test/search_spec.rb
+++ b/Library/Homebrew/test/search_spec.rb
@@ -2,6 +2,7 @@
 
 require "search"
 require "descriptions"
+require "cmd/desc"
 
 RSpec.describe Homebrew::Search do
   describe "#query_regexp" do
@@ -60,7 +61,7 @@ RSpec.describe Homebrew::Search do
   end
 
   describe "#search_descriptions" do
-    let(:args) { instance_double("args") }
+    let(:args) { Homebrew::Cmd::Desc.new(["min_arg_placeholder"]).args }
 
     context "with api" do
       def api_formulae
@@ -72,7 +73,6 @@ RSpec.describe Homebrew::Search do
       end
 
       before do
-        allow(args).to receive_messages(formula?: false, cask?: false, eval_all?: false)
         allow(Homebrew::API::Formula).to receive(:all_formulae).and_return(api_formulae)
         allow(Homebrew::API::Cask).to receive(:all_casks).and_return(api_casks)
       end
```

@dduugg Do you have any better ideas here?

Reply:
I committed this, since it works and I don't need to mock multiple argument combinations.

Reply:
I thought you could use KV pairs to implement instance methods, e.g. `let(:args) { instance_double(Homebrew::CLI::Args, "args", formula?: true) }` or something.

OpenStruct is a bad idea even in tests because it will fail open, e.g. `OpenStruct.new.not_a_method`, while `instance_double("args").not_a_method` will. 

(Apologies if i haven't captured the question, i see this went in, and thus I didn't spend much time gathering context.)

Reply:
Yeah, that probably would have worked too. Good point.